flink-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/JDBCTableSource.java

```diff
@@ -180,6 +193,20 @@ public boolean equals(Object o) {
 		}
 	}
 
+	@Override
+	public String explainSource() {
+		if (selectFields == null) {
+			return String.format(
+				"JDBCTableSource(read fields: %s)", String.join(", ", schema.getFieldNames()));
+		} else {
+			String[] fields = new String[selectFields.length];
+			for (int i = 0; i < selectFields.length; i++) {
+				fields[i] = schema.getFieldName(selectFields[i]).get();
+			}
+			return String.format("JDBCTableSource(read fields: %s)", String.join(", ", fields));
```

Comment:
Use `returnType.getFieldNames` instead reconstruct the selected field names again?

Reply:
Sure